src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/SparqlIndexer.scala

```diff
@@ -0,0 +1,83 @@
+package ch.epfl.bluebrain.nexus.kg.indexing
+
+import akka.http.scaladsl.model.Uri
+import cats.MonadError
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.show._
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient
+import ch.epfl.bluebrain.nexus.commons.sparql.client.SparqlClient
+import ch.epfl.bluebrain.nexus.kg.resolve.Resolution
+import ch.epfl.bluebrain.nexus.kg.resources.Rejection.NotFound
+import ch.epfl.bluebrain.nexus.kg.resources.Resources._
+import ch.epfl.bluebrain.nexus.kg.resources._
+import ch.epfl.bluebrain.nexus.rdf.Iri
+import ch.epfl.bluebrain.nexus.rdf.akka.iri._
+import org.apache.jena.query.ResultSet
+
+import scala.util.Try
+
+/**
+  * Indexer which takes a resource event and calls SPARQL client with relevant update if required
+  *
+  * @param client SPARQL client
+  */
+class SparqlIndexer[F[_]: Resolution](client: SparqlClient[F])(implicit repo: Repo[F],
+                                                               F: MonadError[F, Throwable],
+                                                               ucl: HttpClient[F, ResultSet]) {
```

Comment:
is the HTTP client used? I was thinking that the sparql client is self contained.

Reply:
When making the call to `queryRs`, the `HttpClient[F, ResultSet]` it is used